__tests__/functions.test.js

```diff
@@ -0,0 +1,97 @@
+const functions = require('../functions');
+
+//Testing for the add function
+describe('testing add function', () => {
+    test('add 2 positive numbers', () => {
+        expect(functions.add(2,2)).toBe(4);
+    });
+
+    test('add numbers correctly', () => {
+        expect(functions.add(2,2)).not.toBe(1);
+    });
+
+    test('add 2 negative numbers', () => {
+        expect(functions.add(-2,-2)).toBe(-4);
+    });
+
+    test('add numbers with 0', () => {
+        expect(functions.add(0,1)).toBe(1);
+    });
+
+    test('add 2 nulls', () => {
+        expect(functions.add(null,null)).toBeFalsy();
+    });
+
+    test('add a null and a number', () => {
+        expect(functions.add(null,2)).toBeFalsy();
+    });
+});
+
+//Testing for the multiplication function
+describe('testing multiplication function', () => {
+    test('Multiply 2 positivews', () => {
+        expect(functions.mult(2,4)).toBe(8);
+    });
+
+    test('Multiply 2 negatives', () => {
+        expect(functions.mult(-2,-9)).toBe(18);
+    });
+
+    test('Multiply a positive and a negative', () => {
+        expect(functions.mult(-4,5)).toBe(-20);
+    });
+
+    test('Multiply with 0', () => {
+        expect(functions.mult(0,1)).toBe(0);
+    });
+});
+
+//Testing createUser function
+describe('testing createUser function', () => {
+    test('Should return user name: Mike Wood', () => {
+        expect(functions.createUser()).toEqual({ firstName: 'Mike', lastName: 'Wood' });
+    });
+});
+
+//Testing fetchUser function 
+describe('testing fetchUser function', () => {
+    test('User name should be Leanne Graham', () => {
+        expect.assertions(1);
+        return functions.fetchUser()
+            .then(data => {
+                expect(data.name).toEqual('Leanne Graham');
+            })
+    });
+
+    test('Using async wait to get Leanne Graham', async() => {
+        expect.assertions(1);
+        const data = await functions.fetchUser();
+        expect(data.name).toEqual('Leanne Graham');
+    });
+
+//Testing removeKNames function in various situation
+describe('remove names start with K', () => { 
```

Comment:
Is this whole ```describe``` block is inside of another ```describe``` block testing different function? please move it outside as all tests before